pkg/sql/sem/builtins/builtins.go

```diff
@@ -5579,6 +5583,38 @@ var jsonExtractPathImpl = tree.Overload{
 	Volatility: tree.VolatilityImmutable,
 }
 
+var jsonExtractPathTextImpl = tree.Overload{
+	Types:      tree.VariadicType{FixedTypes: []*types.T{types.Jsonb}, VarType: types.String},
+	ReturnType: tree.FixedReturnType(types.String),
+	Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+		j := tree.MustBeDJSON(args[0])
+		path := make([]string, len(args)-1)
+		for i, v := range args {
+			if i == 0 {
+				continue
+			}
+			if v == tree.DNull {
+				return tree.DNull, nil
+			}
+			path[i-1] = string(tree.MustBeDString(v))
+		}
+		result, err := json.FetchPath(j.JSON, path)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
can you extract lines 5591-5604 into a function that is shared between jsonExtractPathTextImpl and jsonExtractPathImpl?

Reply:
right! missed it 😅